### PR TITLE
[Enhancement] add clang-tidy config && add C++ Code Diagnostic document

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes
add clang-tidy config && add C++ Code Diagnostic document

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dataroaring** [comment]
Could tidy find the bug like below?
https://github.com/apache/incubator-doris/pull/8641

File: docs/en/developer-guide/cpp-diagnostic-code.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "--background-index",
+        "--clang-tidy", //enable clang-tidy
+        "--compile-commands-dir=doris/be/build_RELEASE/", //clangd should read compile_commands.json create by cmake, so you should compile once
+        "--completion-style=detailed",
+        "-j=5", //clangd diagnostic parallelism
+        "--all-scopes-completion",
+        "--pch-storage=memory",
+        "--pretty",
+        "-log=verbose",
+        "--query-driver=ldb_toolchain/bin/g++" //path of g++ or clang++
```
Comment (adonis0147) [inline docs/en/developer-guide/cpp-diagnostic-code.md:50]:
```suggestion
        "--query-driver=ldb_toolchain/bin/*" // path of compiler
```
Using wildcard to match is better.

File: docs/zh-CN/developer-guide/cpp-diagnostic-code.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        "--background-index",
+        "--clang-tidy", //开启clang-tidy
+        "--compile-commands-dir=doris/be/build_RELEASE/", //会用到cmake生成的compile_commands.json,所以需要先编译一次生成该文件
+        "--completion-style=detailed",
+        "-j=5", //clangd分析文件的并行数
+        "--all-scopes-completion",
+        "--pch-storage=memory",
+        "--pretty",
+        "-log=verbose",
+        "--query-driver=ldb_toolchain/bin/g++" //g++或者clang++的路径
```
Comment (adonis0147) [inline docs/zh-CN/developer-guide/cpp-diagnostic-code.md:50]:
```suggestion
        "--query-driver=ldb_toolchain/bin/*" // 编译器的路径
```
Using wildcard to match is better.

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Comment by BiteTheDDDDt** [comment]
> Could tidy find the bug like below? #8641

This new-delet-mismatch that is not in the same scope is not easy to analyze, `-Wmismatched-new-delete` does not work for this case, maybe `scan-build` or `Valgrind` can work.

**Comment by adonis0147** [comment]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**